subprojects/snapshots/src/test/groovy/org/gradle/internal/snapshot/impl/DefaultFileSystemSnapshotterTest.groovy

```diff
@@ -243,6 +249,123 @@ class DefaultFileSystemSnapshotterTest extends Specification {
         })
     }
 
+    def "snapshots a singletonFileTree as RegularFileSnapshot"() {
```

Comment:
Coverage for `zipTree` and `tarTree` is still missing. Snapshotting a `zipTree` works by snapshotting the backing `zip` file. We should have coverage for that here. Something like:
```
def snapshots = snapshotter.snapshot(fileOperations.zipTree(<singleZipFile>))
assertSingleFileSnapshot(snapshots)
```

We also should make sure that snapshotting a `tarTree` without any backing file (from a resource) is snapshotted correctly as a directory file tree.

Reply:
is there a util classes for creating zips in tests? the operation above expects an existing zip file afaik. By no baking file you basically mean an empty tar?

Reply:
`TestFile.zipTo` got you covered. There is also `TestFile.tarTo`.

Reply:
done